services/management/backup/artifacts_service.go

```diff
@@ -96,6 +100,119 @@ func (s *ArtifactsService) ListArtifacts(context.Context, *backupv1beta1.ListArt
 	}, nil
 }
 
+func (s *ArtifactsService) beginDeletingArtifact(
+	artifactID string,
+) (string, *models.S3LocationConfig, error) {
+	var s3Config *models.S3LocationConfig
+	var artifactName string
+	if err := s.db.InTransaction(func(tx *reform.TX) error {
+		artifact, err := models.FindArtifactByID(tx.Querier, artifactID)
+		switch {
+		case err == nil:
+		case errors.Is(err, models.ErrNotFound):
+			return status.Errorf(codes.NotFound, "Artifact with ID %q not found.", artifactID)
+		default:
+			return err
+		}
+		if artifact.Status != models.SuccessBackupStatus &&
```

Comment:
switch on artifact status and on default return err?

Reply:
done